base/range.jl

```diff
@@ -860,13 +860,18 @@ end
 
 median(r::Range{<:Real}) = mean(r)
 
-function in(x, r::Range)
+function _in_range(x, r::Range)
     n = step(r) == 0 ? 1 : round(Integer,(x-first(r))/step(r))+1
     n >= 1 && n <= length(r) && r[n] == x
 end
+in(x::Real, r::Range{<:Real}) = _in_range(x, r)
+# This method needs to be defined separately since -(::T, ::T) can be implemented
+# even if -(::T, ::Real) is not
+in(x::T, r::Range{<:T}) where {T} = _in_range(x, r)
```

Comment:
@JeffBezanson What do you think about using `Range{<:T}` rather than `Range{T}` just to prevent ambiguities here (see [my previous comment](https://github.com/JuliaLang/julia/pull/21728#discussion_r115790401))? This feels like a hack to me and I'd be inclined to remove that method even if that forces people implementing non-`Real` types which support ranges to define it manually.